src/ranch.erl

```diff
@@ -291,16 +291,28 @@ set_max_connections(Ref, MaxConnections) ->
 get_transport_options(Ref) ->
 	ranch_server:get_transport_options(Ref).
 
--spec set_transport_options(ref(), opts()) -> ok | {error, running}.
+-spec set_transport_options(ref(), opts()) -> ok | {error, term()}.
 set_transport_options(Ref, TransOpts0) ->
 	TransOpts = normalize_opts(TransOpts0),
-	case get_status(Ref) of
-		suspended ->
-			ok = ranch_server:set_transport_options(Ref, TransOpts);
-		running ->
-			{error, running}
+	case validate_transport_opts(TransOpts) of
+		ok ->
+			ok = ranch_server:set_transport_options(Ref, TransOpts),
+			ok = apply_transport_options(Ref, TransOpts);
+		TransOptsError ->
+			TransOptsError
 	end.
 
+apply_transport_options(Ref, TransOpts) ->
+	MaxConns = maps:get(max_connections, TransOpts, 1024),
+	HandshakeTimeout = maps:get(handshake_timeout, TransOpts, 5000),
+	Shutdown = maps:get(shutdown, TransOpts, 5000),
+	set_max_connections(Ref, MaxConns),
+	_ = [begin
+		ConnsSup ! {set_handshake_timeout, HandshakeTimeout},
+		ConnsSup ! {set_shutdown, Shutdown}
```

Comment:
Make one `{set_options, TransOpts}` message. Maybe don't bother sending messages if the options didn't change.

Reply:
There is already `set_opts` for changing protocol options, so I'll name the message for transport options `set-transopts`, ok?

Reply:
Then just have `set_transport_options` and `set_protocol_options`.